MIGRATING.md

```diff
@@ -0,0 +1,87 @@
+# Migrating from `v0.4.x` to `v0.5.x`
+
+This guide provides detailed instructions on how to run the built-in converter to
+translate Surface `v0.4` code into the new `v0.5`syntax.
+
+## Limitations of the converter
+
+  * By design, the converter doesn't touch Surface code inside documentation or macro components. If you have
+  any code written inside `<!-- -->` or `<#Raw>...</#Raw>`, you need to convert it manually.
+
+  * The replacement of `~H` with `~F` happens globally in a `.ex` (or `.exs`) file, i.e., the converter will
+  replace any occurrence of `~H` followed by `"""`, `"`, `[`, `(` or `{`, including occurrences found in comments.
+
+  * Running the converter on a project that has already been converted may generate invalid code. If anything goes
+  wrong with the conversion, make sure you revert the changes before running it again.
+
+## Before converting the project
+
+  1. Make sure you have commited your work or have a proper backup before running the converter. It may touch
+  a lot of files so it's recommended to have a safe way to rollback the changes in case anything goes wrong.
+
+  2. Check your dependencies. For a successful migration, all dependencies providing Surface components should also
+  be converted before running the converter. Otherwise, you might not be able to compile your project in case any
+  of those dependencies is using the invalid old syntax. If the dependency you need has not been updated yet,
+  please consider running the converter against it and submitting a PR with updated code. The steps to convert a
+  dependency are the same decribed in this guide.
+
+## Steps to run the converter
+
+Update your `.formatter` informing about `.sface` files and any additional folder where you might have any component
+to be converted:
+
+```
+[
+  surface_inputs: ["{lib,test,priv/catalogue}/**/*.{ex,exs,sface}"],
```

Comment:
Adding this line in my `.formatter.exs` it results in the following error when I run the converter:

```
** (ErlangError) Erlang error: {:badpattern, :missing_delimiter}
    (stdlib 3.15) filelib.erl:63: :filelib.wildcard/2
    (elixir 1.12.0) lib/path.ex:655: Path.wildcard/2
    lib/mix/tasks/surface/convert_syntax.ex:184: anonymous fn/4 in Mix.Tasks.Surface.ConvertSyntax.expand_dot_inputs/2
    (elixir 1.12.0) lib/enum.ex:2356: Enum."-reduce/3-lists^foldl/2-0-"/3
    lib/mix/tasks/surface/convert_syntax.ex:183: Mix.Tasks.Surface.ConvertSyntax.expand_dot_inputs/2
    lib/mix/tasks/surface/convert_syntax.ex:155: Mix.Tasks.Surface.ConvertSyntax.expand_args/3
    lib/mix/tasks/surface/convert_syntax.ex:98: Mix.Tasks.Surface.ConvertSyntax.run/1
    (mix 1.12.0) lib/mix/task.ex:394: anonymous fn/3 in Mix.Task.run_task/3
```

Using the following command works as expected:
```
mix surface.convert_syntax "lib/**/*.{ex,exs,sface}" 
```

Reply:
The issue is with the `priv/catalogue`. If you remove it, it works.

I do not know how to escape `/` inside `{}`, but the following proposition works
```suggestion
  surface_inputs: ["{lib,test,}/**/*.{ex,exs,sface}", "priv/catalogue/**/*.{ex,exs,sface}"],
```